src/vrct/vrctFs/vrct.go

```diff
@@ -0,0 +1,280 @@
+}
+
+const (
+	ConfigFile = iota
+	TextFile
+)
+
+type FilePrototype struct {
+	Layers         []PrototypeLayer
+	RealFileExists bool
+	FileType       *int
+	SelfPath       string `bson:"-"`
+}
+
+type PrototypeLayer struct {
+	// If content path is specified and file exists in real fs, real file will be later overridden by this content
+	// (We don't store content as string in order to make bson lightweight and fast accessible)
+	ContentPath   *string `bson:",omitempty"`
+	IsOptional    bool
+	ConfigOptions []ConfigOption `bson:",omitempty"`
+	ConfigType    *int           `bson:",omitempty"`
+	// TODO: stack trace
+}
+
+func (p *FilePrototype) mergeLayers(optLayersIndexesToSkip []int) (PrototypeLayer, error) {
+	pl := PrototypeLayer{
+		IsOptional: false,
+	}
+
+	encounteredOptionalLayers := 0
+
+	for i, layer := range p.Layers {
+		if layer.IsOptional {
+			encounteredOptionalLayers++
+			if slices.Contains(optLayersIndexesToSkip, encounteredOptionalLayers-1) {
+				continue
+			}
+		}
+
+		var optionalInfoPrefix string
+		if layer.IsOptional {
+			optionalInfoPrefix = "optional"
+		} else {
+			optionalInfoPrefix = "non optional"
+		}
+		potentialConflictError := fmt.Errorf(
+			"%s layer nr. %d is in conflict with previous layers", optionalInfoPrefix, i,
+		)
+
+		pl.ConfigOptions = append(pl.ConfigOptions, layer.ConfigOptions...)
+		pl.IsOptional = layer.IsOptional && pl.IsOptional
+
+		if pl.ConfigType != nil && layer.ConfigType != nil {
+			return pl, potentialConflictError
+		}
+		pl.ConfigType = layer.ConfigType
+	}
+
+	// Not optional layers first
+	sort.Slice(p.Layers, func(i, j int) bool {
+		return !p.Layers[i].IsOptional && p.Layers[j].IsOptional
+	})
+
+	for i, layer := range p.Layers {
+		if pl.ContentPath != nil && layer.ContentPath != nil && !layer.IsOptional {
+			return pl, fmt.Errorf(
+				"non optional layer nr. %d is in conflict with previous layers", i,
+			)
+		}
+		if (layer.IsOptional && pl.ContentPath == nil) || !layer.IsOptional {
+			pl.ContentPath = layer.ContentPath
+		}
+	}
+
+	return pl, nil
+}
+
+func (p *FilePrototype) getDestinationPath() string {
+	splitPath := strings.Split(p.SelfPath, "/")
+	return strings.Join(splitPath[:len(splitPath)-1], "/")
+}
+
+func (p *FilePrototype) SimulateFile() ([]byte, error) {
+	finalLayer, err := p.mergeLayers([]int{})
+	if err != nil {
+		return nil, err
+	}
+
+	var fileType int
+	if p.FileType == nil {
+		fileType = TextFile
+	} else {
+		fileType = *p.FileType
+	}
+
+	switch fileType {
+	case TextFile:
+		var filePath string
+
+		if p.RealFileExists {
+			filePath = p.getDestinationPath()
+		} else {
+			filePath = *finalLayer.ContentPath
+		}
+
+		file, err := os.ReadFile(filePath)
+		if err != nil {
+			return nil, err
+		}
+
+		return file, nil
+
+	case ConfigFile:
+		// TODO
+		break
+	}
+
+	// Todo: handle other FileTypes
+	panic("Unimplemented simulation FileType simulation")
+}
+
+func (p *FilePrototype) Read(prototypePath string) error {
```

Comment:
Why it doesn't return any content?

Reply:
It writes read value to `FilePrototype` struct